src/vs/base/browser/ui/tree/objectTreeModel.ts

```diff
@@ -182,7 +178,9 @@ export class ObjectTreeModel<T extends NonNullable<any>, TFilterData extends Non
 		const location = this.getElementLocation(element);
 		const node = this.model.getNode(location);
 
-		this._setChildren(location, this.resortChildren(node, recursive));
+		// it should be safe to always use diff mode here if an identity provider
+		// is available, since we know the nodes and their nested children are unchanged.
+		this._setChildren(location, this.resortChildren(node, recursive), { diffIdentityProvider: this.identityProvider });
```

Comment:
This is the risky change which might break other people's trees. We should maybe think about an opt-in strategy for now? Maybe it's OK tho... 😄 We just have to be aware of breakage across the workbench once this gets merged.

Reply:
I think this would only be a problem if there were two nodes with the same identity in the tree--is that even legal?

Otherwise, if identity is provided we know that this operation doesn't involve updating nodes or requiring rerendering... I guess the one issue would be if anyone is using `resort` as a replacement for `rerender` or `setChildren`.

Reply:
I'll leave this out for the moment; I want to get this change in. And resorting should be quite rare anyway.